src/Core/Image/ImageFormatConfiguration.php

```diff
@@ -83,4 +112,12 @@ public function isGenerationFormatSet(string $format): bool
     {
         return in_array($format, $this->getGenerationFormats());
     }
+
+    /**
+     * reset static cache (eg unit testing purpose).
+     */
+    public static function resetStaticCache()
```

Comment:
Ah you probably already noticed this cache issue in the tests 😅
I'm not sure it will do exactly what you intend to but it seems to me a regular class field would have the same effect for your optimization without the drawbacks of static cache and this method would not be needed any more